src/System.Private.CoreLib/src/System/Threading/EventWaitHandle.cs

```diff
@@ -90,7 +90,7 @@ public EventWaitHandle(bool initialState, EventResetMode mode, string name, out
             };
 
             IntPtr unsafeHandle = Interop.mincore.CreateEventEx(IntPtr.Zero, name, eventFlags, (uint)Interop.Constants.EventAllAccess);
-            int errorCode = (int)Interop.mincore.GetLastError();
+            int errorCode = Marshal.GetLastWin32Error();
```

Comment:
CreateEventEx is not marked with SetLastError=true, so this won't work.

It may be a good idea to do these cleanup changes in a follow up change on ProjectN/TFS side where you can test them well.